hubgrep/lib/hosting_service_interfaces/_hosting_service_interface.py

```diff
@@ -88,33 +87,37 @@ def __init__(
         self,
         host_service_id,
         api_url,
-        label,
         search_path,
+        label,
+        config_dict,
         cached_session: CachedSession,
         timeout=None,
     ):
-
         self.host_service_id = host_service_id
         self.api_url = api_url
         self.label = label
+        self.config_dict = config_dict
         self.request_url = urljoin(self.api_url, search_path)
         self.timeout = timeout
         self.cached_session = cached_session
         self.cached_session.headers.update({"referer": current_app.config["REFERER"]})
 
     def search(
-        self, keywords: list = [], tags: dict = {}
-    ) -> ("HostingServiceInterface", CachedResponse, List):
+            self, keywords: list = [], tags: dict = {}
+    ) -> "HostingServiceInterfaceResult":
         time_before = time.time()
-        hosting_service_interface, response_result, results = self._search(
+        hosting_service_interface_result = self._search(
             keywords, tags
         )
         logger.debug(f"search on {self.api_url} took {time.time() - time_before}s")
-        return hosting_service_interface, response_result, results
+        return hosting_service_interface_result
 
-    def _search(self, keywords: list, tags: dict):
+    def _search(self, keywords: list, tags: dict) -> "HostingServiceInterfaceResult":
         raise NotImplementedError
 
+    def _get_request_headers(self) -> dict:
+        return dict()
+
```

Comment:
_get_request_headers

I added this method on the baseclass so that its always there regardless of which service it currently is, and maybe there will be headers in the future that we always want to set - which we then do here.